brayns/engine/scene/ModelsOperations.cpp

```diff
@@ -18,33 +18,41 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include "ExtractComponent.h"
+#include "ModelsOperations.h"
 
-#include <brayns/engine/components/ColorRampComponent.h>
-#include <brayns/engine/components/MaterialComponent.h>
-#include <brayns/engine/components/SimulationComponent.h>
+#include <brayns/engine/components/ClipperViews.h>
+#include <brayns/engine/components/Lights.h>
 
 namespace brayns
 {
-Material &ExtractComponent::material(Model &model)
+void ModelsOperations::removeLights(ModelManager &models)
 {
-    auto &component = model.getComponent<MaterialComponent>();
-    return component.getMaterial();
+    removeModelsWithComponent<Lights>(models);
 }
 
-ColorRamp &ExtractComponent::colorRamp(Model &model)
+void ModelsOperations::removeClippers(ModelManager &models)
 {
-    auto &component = model.getComponent<ColorRampComponent>();
-    return component.getColorRamp();
+    removeModelsWithComponent<ClipperViews>(models);
 }
 
-bool ExtractComponent::simulationEnabled(Model &model)
+void ModelsOperations::removeObjects(ModelManager &models)
 {
-    auto component = model.findComponent<SimulationComponent>();
-    if (!component || !component->enabled())
+    auto &instances = models.getAllModelInstances();
+
+    std::vector<uint32_t> ids;
+    ids.reserve(instances.size());
+
+    for (auto instance : instances)
     {
-        return false;
+        auto &model = instance->getModel();
+        auto &components = model.getComponents();
+        if (components.has<Lights>() || components.has<ClipperViews>())
```

Comment:
It will not scale very well if we need to add new non-renderable types...

Could it be something like removeModelsWithComponent<Renderable>(models); ?

Reply:
Addressed